srv/modules/runners/validate.py

```diff
@@ -170,7 +170,7 @@ def __init__(self, name, data=None, grains=None, printer=None):
     def __dev_env(self):
         if 'DEV_ENV' in os.environ:
             return os.environ['DEV_ENV'].lower() != 'false'
-        elif len(self.data.keys()) > 1:
+        elif len(self.data.keys()) >= 1:
```

Comment:
change this line to: `elif self.data:`

This will test if the `self.data` is not empty, which is a more correct (pythonic) way than checking if the number of keys is greater than 0.

Reply:
Thanks for the feeback! As I am not a real developer - I do not care how this is fixed ;-)
>= works for be - but if someone can rest it I would also be fine with elif self.data: .

The major question I have is more about "why do we need to test for existing minions here at all?"

Reply:
It's just because the way the pillar data is being assigned to that `self.data` variable. the `self.data` variable is assigned with the output of the `salt '*' pillar.items` salt command, which corresponds to a dictionary where each key is the `minion_id` and the value is the pillar data.

Long story short, technical reasons.